src/shogun/evaluation/Calibration.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2011-2012 Heiko Strathmann
+ * Copyright (C) 2011 Berlin Institute of Technology and Max-Planck-Society
+ */
+
+#include <shogun/evaluation/Calibration.h>
+#include <shogun/evaluation/CalibrationMethod.h>
+#include <shogun/lib/config.h>
+#include <shogun/machine/Machine.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/Statistics.h>
+
+using namespace shogun;
+
+CBinaryLabels* CCalibration::apply_binary(CFeatures* features)
+{
+
+	CBinaryLabels* result_labels = (CBinaryLabels*)m_machine->apply(features);
```

Comment:
this explicit casting will cause a lot of trouble if the returned label is not binary of the machine :) you should either do a dynamic_cast to check for this or simply `get_label_type() == LT_BINARY`

Reply:
Should I use SG_ERROR to indicate that the machine should return binary labels when the labels are not binary?

Reply:
you could use REQUIRE

Reply:
Understood

Reply:
Hello @vigsterkr, I have used the approach that you have mentioned (using `get_label_type` and `REQUIRE`). However, shogun seems to have a LabelsFactory class which has methods for doing the same thing. Is that the recommended way of doing this?

Reply:
mmm oh yeah labelfactory, indeed its the best option